climpred/utils.py

```diff
@@ -97,3 +103,52 @@ def reduce_time_series(forecast, reference, nlags):
     forecast = forecast.where(forecast.time >= imin, drop=True)
     reference = reference.where(reference.time >= imin, drop=True)
     return forecast, reference
+
+
+def assign_climpred_compute_to_attrs(
+    skill,
+    ds,
+    function_name='compute_perfect_model',
+    metadata_dict=None,
+    metric=None,
+    comparison=None,
+):
+    """Write information about prediction skill into attrs."""
+    # assign old attrs
+    skill.attrs = ds.attrs
+
+    # climpred info
+    skill.attrs[
+        'Prediction skill'
+    ]: f'calculated by climpred https://climpred.readthedocs.io/'
+    skill.attrs['skill calculated by function'] = function_name
+    if 'init' in ds.coords:
+        skill.attrs['number of initializations'] = ds.init.size
+    if 'member' in ds.coords:
+        skill.attrs['number of members'] = ds.member.size
+
+    if 'perfect_model' in function_name:
+        comparison = get_comparison_function(comparison, PM_COMPARISONS).__name__[1:]
+        metric = get_metric_function(metric, PM_METRICS).__name__[1:]
+    elif 'hindcast' in function_name:
+        comparison = get_comparison_function(comparison, HINDCAST_COMPARISONS).__name__[
+            1:
+        ]
+        metric = get_metric_function(metric, HINDCAST_METRICS).__name__[1:]
+
+    skill.attrs['metric'] = metric
+    skill.attrs['comparison'] = comparison
+
+    if metric in ['pearson_r']:
+        skill.attrs['units'] = ' '
```

Comment:
I prefer not to have a space as a unit; when it's printed out people assume it's '' and when they do some comparison like

```
if ds.attrs['units'] == '':
    print('units')
```
they'll probably have to debug very long to realize that `units == ' '`

Reply:
I don't fully understand what's going on here. Could you explain, and if we need it, add a comment for clarity here? Why is units being added as a blank attribute? Also on a technical point, you should add the other metric keywords to this list, unless it's being converted to the main `pearson_r` keyword upon entering this function.

Reply:
ACC is unitless, therefore I set it to no unit - here `' '`. RMSE would have the same units, where I dont overwrite anything. the keyword str metric that is written to attrs comes from `get_metric_function` and therefore from `HINDCAST/PM_METRICS`. these lists are the lists of all metrics without ALIASES. sure a few other metrics should also can units. I'll change that

Reply:
@aaronspring, I think a comprimise would be to do `ds.attrs['units'] = 'None'` so that it is very explicit that we expect there to be no units without the issue of just a space.

Reply:
Also don't some other metrics expect no units? From a quick metric look:

* NMSE (since it's normalized)
* NMAE
* NRMSE
* MSSS
* std ratio
* bias slope
* conditional bias
* msss_murphy
* LESS
* some of the probabilistic...

Maybe the normalized ones and skill scores have some other units but they're definitely not the original units of the variables. But there's definitely some nuanced unit things going on here. Some get square rooted so it should actually be something like `ds.attrs['units'] = 'square root of ' + orig_units`. It might be hard to expect/modify string units directly to account for raising to powers and so on.

Reply:
fine with None.